src/ray/gcs/gcs_server/gcs_server.cc

```diff
@@ -515,24 +525,27 @@ void GcsServer::CollectStats() {
       (RayConfig::instance().metrics_report_interval_ms() / 2) /* milliseconds */);
 }
 
-void GcsServer::PrintDebugInfo() {
+void GcsServer::DumpDebugStateToFile() const {
```

Comment:
hmm I think it shouldn't be GcsServer's responsibility to dump state into files. Can you reuse the even framework or something similar?

Reply:
Hmm This is how it works in raylet now.

And I don't think event framework is a good idea to use here (it might be too much noise).

Is your requirement that this file dump should happen in other self-contained module?

Reply:
Hmm maybe we can do something like

```
    ray::RayEventInit(ray::rpc::Event_SourceType::Event_SourceType_GCS,
                      std::unordered_map<std::string, std::string>(), log_dir,
                      RayConfig::instance().event_level(),
                      /*debug_dump*/{gcs_server_});
```
?

Reply:
Or ray::DebugDumpManager({gcs_server_, main_service})

Reply:
alright.. maybe add a TODO to fix it later